src/inc/configuration-document.ts

```diff
@@ -0,0 +1,116 @@
+import defu from 'defu'
+import type { Scheme, OpenIDConnectConfigurationDocument } from '../index'
+import Storage from '../core/storage'
+
+class ConfigurationDocumentRequestError extends Error {
+  constructor() {
+    super('Error loading OpenIDConnect configuration document')
+    this.name = 'ConfigurationDocumentRequestError'
+  }
+}
+
+// eslint-disable-next-line no-console
+const ConfigurationDocumentWarning = (message: string) =>
+  // eslint-disable-next-line no-console
+  console.warn(`[AUTH] [OPENID CONNECT] Invalid configuration. ${message}`)
+
+export default class ConfigurationDocument {
+  public scheme: Scheme
+  public $storage: Storage
+  public key: string
+
+  constructor(scheme: Scheme, storage: Storage) {
+    this.scheme = scheme
+    this.$storage = storage
+    this.key = '_configuration_document.' + this.scheme.name
+  }
+
+  _set(value: OpenIDConnectConfigurationDocument | boolean) {
+    return this.$storage.setState(this.key, value)
+  }
+
+  get(): OpenIDConnectConfigurationDocument {
+    return this.$storage.getState(this.key)
```

Comment:
Same here. Use `getUniversal` instead.

Reply:
I've implemented `setState` in favor of `setUniversal` because I didn't want to save the configuration document, which can contain a fairly large amount of data for a cookie, in a cookie. This prevents that requests headers will become to large.

The configuration-document of the demo for example:
https://gist.github.com/vanhoofmaarten/fc1ecb5bb59a7bcd9364e6233fffb8df

In universal mode, I chose to fetch and set the configuration document state on server and use the `beforeNuxtRender` method to prepare the document for hydration to the client.

How shall I continue?